staging/src/k8s.io/apiserver/pkg/storage/cacher/cacher_whitebox_test.go

```diff
@@ -2470,6 +2470,72 @@ func TestWatchStreamSeparation(t *testing.T) {
 	}
 }
 
+func TestDispatchEventsUseProgressRequester(t *testing.T) {
```

Comment:
The test works but can start flaking on "timing" issues. I like it because it test the entire watch request.